src/np_codeocean/upload.py

```diff
@@ -355,19 +366,50 @@ def create_codeocean_upload(session: str | int | np_session.Session,
     if upload.behavior_videos:
         create_behavior_videos_symlinks(upload.session, upload.behavior_videos)
 
-    try:
-        dynamic_routing_task.add_rig_to_session_dir(
-            np_config.normalize_path(root),
-            session.date,
+    session_dir = np_config.normalize_path(root)
+    if modality in ('ecephys', ):
+        logger.debug(
+            "Adding rig metadata for ecephys session. modality=%s"
+            % modality)
+        try:
+            dynamic_routing_task.add_rig_to_session_dir(
+                session_dir,
+                session.date,
+                np_config.normalize_path(
+                    pathlib.Path(CONFIG["rig_metadata_dir"])
+                ),
+            )
+        except Exception:
+            logger.error(
+                "Failed to update session and rig metadata for Code Ocean upload.",
+                exc_info=True,
+            )
+    else:
```

Comment:
this should be `elif session.modality == 'behavior': `

`else:` should raise a ValueError for an unexpected modality name

Reply:
Addressed here: [2db5956](https://github.com/AllenInstitute/np_codeocean/pull/13/commits/2db595654b38665d5297fd4d648d8af9082ccb38)